OTAnalytics/plugin_ui/customtkinter_gui/line_section.py

```diff
@@ -202,14 +205,22 @@ def update(self, coordinate: tuple[int, int], event_type: str) -> None:
             coordinates (tuple[int, int]): Coordinates clicked on canvas
             event_type (str): Event type of canvas click
         """
-        if (not self.geometry_builder.has_start()) and (
-            event_type == "left_mousebutton_up"
-        ):
-            self.geometry_builder.add_coordinate(coordinate)
-        elif self.geometry_builder.has_start() and event_type == "mouse_motion":
-            self.geometry_builder.set_tmp_end(coordinate)
-        elif self.geometry_builder.has_start() and event_type == "left_mousebutton_up":
+        if event_type == "left_mousebutton_up":
+            print("left_mousebutton_up")
             self.geometry_builder.add_coordinate(coordinate)
+        elif (
+            self.geometry_builder.number_of_coordinates() >= 1
+            and event_type == "mouse_motion"
+        ):
+            self.geometry_builder.add_temporary_coordinate(coordinate)
+        elif self.geometry_builder.number_of_coordinates() >= 2 and (
+            event_type in {"right_mousebutton_up", "return"}
```

Comment:
Why a dictionary and not a list: `[]` or tuple: `()` ?

Reply:
Given the definition of a set, it seems to be behave a dict without values :)

Reply:
That's right, a set is in python like a dict without values. I have to lower by Java skills for this one ;-)